CreateProcess failed: The system cannot find the file specified. returned while compiling on Windows

Thanks for reporting the unexpected results and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmcv.readthedocs.io/en/latest/trouble_shooting.html) but cannot get the expected help.
3. The unexpected results still exist in the latest version.

**Describe the Issue**
A clear and concise description of what the bug is, including what results are expected and what the real results you got.

**Reproduction**

1. What command, code, or script did you run?

```bash
A placeholder for the command.
```

2. Did you make any modifications on the code? Did you understand what you have modified?

**Environment**

1. Please run `python -c "from mmcv.utils import collect_env; print(collect_env())"` to collect necessary environment information and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
```
10:56:02  FAILED: D:/Workspace/Jenkins/workspace/mmcv/build-on-windows-commit@2/mmcv/build/temp.win-amd64-3.7/Release/mmcv/ops/csrc/pytorch/deform_conv.obj 
10:56:02  cl /showIncludes /nologo /Ox /W3 /GL /DNDEBUG /MD /MD /wd4819 /EHsc -DMMCV_WITH_CUDA -ID:\Workspace\Jenkins\workspace\mmcv\build-on-windows-commit@2\mmcv\mmcv\ops\csrc\common -ID:\Workspace\Jenkins\workspace\mmcv\build-on-windows-commit@2\mmcv\mmcv\ops\csrc\common\cuda -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\include -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\include\torch\csrc\api\include -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\include\TH -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\include\THC "-ID:\NVIDIA GPU Computing Toolkit\CUDA\v10.2\include" -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\include -ID:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt" -c D:\Workspace\Jenkins\workspace\mmcv\build-on-windows-commit@2\mmcv\mmcv\ops\csrc\pytorch\deform_conv.cpp /FoD:\Workspace\Jenkins\workspace\mmcv\build-on-windows-commit@2\mmcv\build\temp.win-amd64-3.7\Release\mmcv/ops/csrc/pytorch\deform_conv.obj -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 /std:c++14
10:56:02  CreateProcess failed: The system cannot find the file specified.
10:56:02  ninja: build stopped: subcommand failed.
10:56:02  Traceback (most recent call last):
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\utils\cpp_extension.py", line 1400, in _run_ninja_build
10:56:02      check=True)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\subprocess.py", line 512, in run
10:56:02      output=stdout, stderr=stderr)
10:56:02  subprocess.CalledProcessError: Command '['ninja', '-v', '-j', '8']' returned non-zero exit status 1.
10:56:02  
10:56:02  During handling of the above exception, another exception occurred:
10:56:02  
10:56:02  Traceback (most recent call last):
10:56:02    File "setup.py", line 391, in <module>
10:56:02      zip_safe=False)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\setuptools\__init__.py", line 153, in setup
10:56:02      return distutils.core.setup(**attrs)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\core.py", line 148, in setup
10:56:02      dist.run_commands()
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\dist.py", line 966, in run_commands
10:56:02      self.run_command(cmd)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\dist.py", line 985, in run_command
10:56:02      cmd_obj.run()
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\setuptools\command\build_ext.py", line 79, in run
10:56:02      _build_ext.run(self)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\command\build_ext.py", line 340, in run
10:56:02      self.build_extensions()
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\utils\cpp_extension.py", line 580, in build_extensions
10:56:02      build_ext.build_extensions(self)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\command\build_ext.py", line 449, in build_extensions
10:56:02      self._build_extensions_serial()
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\command\build_ext.py", line 474, in _build_extensions_serial
10:56:02      self.build_extension(ext)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\setuptools\command\build_ext.py", line 202, in build_extension
10:56:02      _build_ext.build_extension(self, ext)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\distutils\command\build_ext.py", line 534, in build_extension
10:56:02      depends=ext.depends)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\utils\cpp_extension.py", line 562, in win_wrap_ninja_compile
10:56:02      with_cuda=with_cuda)
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\utils\cpp_extension.py", line 1140, in _write_ninja_file_and_compile_objects
10:56:02      error_prefix='Error compiling objects for extension')
10:56:02    File "D:\miniconda3\envs\mmcvv1.4.6_cuda102_py37_torch1.5.0\lib\site-packages\torch\utils\cpp_extension.py", line 1413, in _run_ninja_build
10:56:02      raise RuntimeError(message)
10:56:02  RuntimeError: Error compiling objects for extension
10:56:04  Start python setup.py build_ext.
10:56:04  Python setup.py build_ext failed.
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -111,8 +111,15 @@ jobs:
       # pstuil is an optional package to detect the number of CPU for compiling mmcv
       - name: Install psutil
         run: pip install psutil
-      - name: Build and install
-        run: rm -rf .eggs && pip install -e .
+      - name: Create sdist and untar
+        run: |
+          MMCV_WITH_OPS=1 python setup.py sdist
+          tar zxvf dist/mmcv-full* -C /tmp
+      - name: Build and install from sdist
+        run: |
+          pushd /tmp/mmcv-full*
+          pip install -e .
+          popd
       - name: Validate the installation
         run: python -c "import mmcv"
       - name: Run unittests and generate coverage report

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -115,6 +115,7 @@ jobs:
         run: |
           MMCV_WITH_OPS=1 python setup.py sdist
           tar zxvf dist/mmcv-full* -C /tmp
+          rm -r mmcv
       - name: Build and install from sdist
         run: |
           pushd /tmp/mmcv-full*

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,6 +1,6 @@
 include requirements/runtime.txt
 include mmcv/model_zoo/open_mmlab.json mmcv/model_zoo/deprecated.json mmcv/model_zoo/mmcls.json
 include mmcv/ops/csrc/common/cuda/*.cuh mmcv/ops/csrc/common/cuda/*.hpp mmcv/ops/csrc/common/*.hpp
-recursive-include mmcv/ops/csrc/common/utils/spconv *.h
 include mmcv/ops/csrc/pytorch/*.cpp mmcv/ops/csrc/pytorch/cuda/*.cu mmcv/ops/csrc/pytorch/cuda/*.cpp mmcv/ops/csrc/pytorch/cpu/*.cpp
 include mmcv/ops/csrc/parrots/*.h mmcv/ops/csrc/parrots/*.cpp
+recursive-include mmcv/ops/csrc/common/utils/spconv *.h *.hpp *.cpp *.cuh *.cu

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -3,4 +3,4 @@ include mmcv/model_zoo/open_mmlab.json mmcv/model_zoo/deprecated.json mmcv/model
 include mmcv/ops/csrc/common/cuda/*.cuh mmcv/ops/csrc/common/cuda/*.hpp mmcv/ops/csrc/common/*.hpp
 include mmcv/ops/csrc/pytorch/*.cpp mmcv/ops/csrc/pytorch/cuda/*.cu mmcv/ops/csrc/pytorch/cuda/*.cpp mmcv/ops/csrc/pytorch/cpu/*.cpp
 include mmcv/ops/csrc/parrots/*.h mmcv/ops/csrc/parrots/*.cpp
-recursive-include mmcv/ops/csrc/common/utils/spconv *.h *.hpp *.cpp *.cuh *.cu
+recursive-include mmcv/ops/csrc/ *.h *.hpp *.cpp *.cuh *.cu

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,5 +1,6 @@
 include requirements/runtime.txt
 include mmcv/model_zoo/open_mmlab.json mmcv/model_zoo/deprecated.json mmcv/model_zoo/mmcls.json
 include mmcv/ops/csrc/common/cuda/*.cuh mmcv/ops/csrc/common/cuda/*.hpp mmcv/ops/csrc/common/*.hpp
+recursive-include mmcv/ops/csrc/common/utils/spconv *.h
 include mmcv/ops/csrc/pytorch/*.cpp mmcv/ops/csrc/pytorch/cuda/*.cu mmcv/ops/csrc/pytorch/cuda/*.cpp mmcv/ops/csrc/pytorch/cpu/*.cpp
 include mmcv/ops/csrc/parrots/*.h mmcv/ops/csrc/parrots/*.cpp

